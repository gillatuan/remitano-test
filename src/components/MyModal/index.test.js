/* eslint-disable testing-library/prefer-screen-queries */
import { MyModal } from './index';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { testSnapshots } from 'test';
import { Modal } from 'react-bootstrap';

const makeSut = (props) => {
  const {
    children,
    dialogClassName = 'my-modal',
    name = 'my-modal',
    show = false,
    size = 'lg',
    title = 'My Modal',
    setShow,
  } = props;
  return render(
    <MyModal
      dialogClassName={dialogClassName}
      name={name}
      show={show}
      size={size}
      title={title}
      setShow={setShow}
    >
      {children}
    </MyModal>
  );
};

const initialConfig = {
  ariaLabel: 'aria-input-title',
  dialogClassName: 'my-modal',
  id: 'input-title',
  name: 'my-modal',
  title: 'My Modal'
};

describe('<MyModal /> props', () => {
  testSnapshots(Modal, [
    {
      props: initialConfig,
      description: 'renders Modal correctly',
    },
    {
      props: {
        ...initialConfig,
        title: 'My Modal'
      },
      description: 'renders Modal with My Modal in title',
    },
  ]);
});
