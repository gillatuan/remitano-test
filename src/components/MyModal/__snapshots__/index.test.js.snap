// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> snapshots renders disable input correctly 1`] = `null`;

exports[`<InputField /> snapshots renders input correctly 1`] = `null`;

exports[`<InputField /> snapshots renders input is checkbox 1`] = `null`;

exports[`<InputField /> snapshots renders input is date 1`] = `null`;

exports[`<InputField /> snapshots renders input is disabled checkbox 1`] = `null`;

exports[`<InputField /> snapshots renders input is disabled radio 1`] = `null`;

exports[`<InputField /> snapshots renders input is email 1`] = `null`;

exports[`<InputField /> snapshots renders input is file 1`] = `null`;

exports[`<InputField /> snapshots renders input is image 1`] = `null`;

exports[`<InputField /> snapshots renders input is number 1`] = `null`;

exports[`<InputField /> snapshots renders input is radio 1`] = `null`;

exports[`<InputField /> snapshots renders input is search 1`] = `null`;

exports[`<InputField /> snapshots renders input is select 1`] = `null`;

exports[`<InputField /> snapshots renders input is select multi 1`] = `null`;

exports[`<InputField /> snapshots renders input is textarea 1`] = `null`;

exports[`<InputField /> snapshots renders readonly input 1`] = `null`;

exports[`<MyModal /> props snapshots renders Modal correctly 1`] = `null`;

exports[`<MyModal /> props snapshots renders Modal with My Modal in title 1`] = `null`;

exports[`<MyModal /> props snapshots renders disable input correctly 1`] = `null`;

exports[`<MyModal /> props snapshots renders input correctly 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is checkbox 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is date 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is disabled checkbox 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is disabled radio 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is email 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is file 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is image 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is number 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is radio 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is search 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is select 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is select multi 1`] = `null`;

exports[`<MyModal /> props snapshots renders input is textarea 1`] = `null`;

exports[`<MyModal /> props snapshots renders readonly input 1`] = `null`;
